app/models/questionnaire_version.rb

```diff
@@ -0,0 +1,29 @@
+class QuestionnaireVersion < ActiveRecord::Base
+  has_one :questionnaire, required: false
+
+  def localized_layout
+    localize_layout(layout)
+  end
+
+  private
+
+  def localize_layout(sub_layout)
+    sub_layout
+      .then { |l| localize_text(l) }
+      .then { |l| localize_items_recursively(l) }
+  end
+
+  def localize_text(sub_layout)
+    text = sub_layout["text"]
+    return sub_layout unless text
+
+    sub_layout.merge({"text" => I18n.t(text)})
```

Comment:
What would happen if we don't have translations for a locale? 

Is it an uncaught exception? What will the API response code be in this case?

Reply:
Currently, missing translations simply read `translation missing: text`.

Adding a `!` to `I18n.t!` will respond with 500 when translations are missing.

Reply:
Another approach that we thought of and discarded: When translations are missing, Server sends a Sentry error.
In the API response, the form with missing translation will be omitted. Response will contain other forms that do have translations.
Issue with this is updated-at cursor would have moved, and there's no way to sync missing translation form unless it's touched in DB.

Reply:
We should not expose missing translations through the api, so the response should not contain `translation missing: text`.

> Issue with this is updated-at cursor would have moved, and there's no way to sync missing translation form unless it's touched in DB.

This is an important point, it's possible that the translations get updated, how do we handle those situations?

Reply:
> the response should not contain `translation missing: text.`
Agreed. That's why we'll respond with 500.

> it's possible that the translations get updated, how do we handle those situations?
In current design, the Questionnaire table's `updated_at` row must be touched to handle this situation.